[SPARK-40968] Fix a few wrong/misleading comments in DAGSchedulerSuite

JiexingLi opened the pull request:
### What changes were proposed in this pull request?
Fix a few wrong or misleading comments in DAGSchedulerSuite. 

### Why are the changes needed?
The wrong or misleading comments in DAGSchedulerSuite introduce confusions and make it harder to understanding the code.

### Does this PR introduce _any_ user-facing change?
No


### How was this patch tested?
No code changes, pure comment changes.
Original tests pass.

JiexingLi committed:
fix some wrong/confusing comments

HyukjinKwon commented:
@JiexingLi please keep the PR desscription template (https://github.com/apache/spark/blob/master/.github/PULL_REQUEST_TEMPLATE) and keep the PR title formtted like others (see also https://spark.apache.org/contributing.html)

AmplabJenkins commented:
Can one of the admins verify this patch?

itholic commented:
@JiexingLi  Can you address https://github.com/apache/spark/pull/38371#issuecomment-1288700659 when you find some time? Just for reminder, take your time.

JiexingLi commented:
Thank @HyukjinKwon and @itholic for the information. I have now updated the PR description and added the ticket for the title. 

Can you help take another look? Thank you so much.

HyukjinKwon commented on core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala:
```diff
@@ -3089,13 +3089,14 @@ class DAGSchedulerSuite extends SparkFunSuite with TempLocalSparkContext with Ti
 
     // Finish the first 2 shuffle map stages.
-    completeShuffleMapStageSuccessfully(0, 0, 2)
+    completeShuffleMapStageSuccessfully(0, 0, 2, Seq("hostA", "hostB"))
```
This not only fixes the comment but the test codes. Mind elaborating this please?

JiexingLi replied:
Yes. The background is that: Wenchen and I were reading the related code, but felt confused about why line 3098 says: "Executor lost on hostB, both of stage 0 and 1 should be reran.". 

After digging around, we found that stage0 completes in "hostA", "hostB" (the default hosts somewhere in the code). But it is not very obvious for reader to tell this.

So adding  Seq("hostA", "hostB") to reader to easily get this. I further updated the comments too.

mridulm replied:
This change is not required.
Fetch failed is due to stage 1 partition on hostB going missing - it does not matter where stage 0 ran - it will be recomputed for indeterminate stages.

JiexingLi replied:
Yes, this is not required. It is only added for better readability (as I mentioned, "hostA", "hostB" are the default hosts). In my opinion, having the value here, we don't need to go to read completeShuffleMapStageSuccessfully(), but only the code here, we can know what happen. Beside, it might be good to keep the two completeShuffleMapStageSuccessfully() consistent here (both having hostNames, or both not having hostNames)?  Let me know if you think I should delete Seq("hostA", "hostB") here. 

I added "In case no hostNames are provided, the tasks will progressively complete on hostA, hostB, etc." to completeShuffleMapStageSuccessfully().

mridulm replied:
Enhancing documentation of `completeShuffleMapStageSuccessfully` is sufficient - we dont want to explicitly specify `hostNames` for all invocations of the method.

JiexingLi replied:
Sounds good. Deleted the hostNames param.

JiexingLi committed:
trigger CIs

JiexingLi committed:
add more comments for executor lost on hostB

mridulm commented on core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala:
```diff
@@ -3089,13 +3089,14 @@ class DAGSchedulerSuite extends SparkFunSuite with TempLocalSparkContext with Ti
 
-    // Executor lost on hostB, both of stage 0 and 1 should be reran.
+    // FetchFailed on stage 2, both of stage 1 and 2 should be reran. Besides, executor lost on
+    // hostB, both of stage 0 and 1 should be reran.
```
This comment change is not required.

JiexingLi replied:
Thanks, Mridul. 

I updated the comment. Your rewrite is very neat and clear.

mridulm commented on core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala:
```diff
@@ -3207,7 +3209,7 @@ class DAGSchedulerSuite extends SparkFunSuite with TempLocalSparkContext with Ti
     val failedStages = scheduler.failedStages.toSeq
     assert(failedStages.length == 2)
-    // Shuffle blocks of "hostA" is lost, so first task of the `shuffleMapRdd2` needs to retry.
+    // Shuffle blocks of "hostA" is lost, so first task of the `finalRdd` needs to retry.
```
`mapRdd` actually.

JiexingLi replied:
Yes, my bad. 

Thanks and updated.

JiexingLi committed:
address comments

JiexingLi committed:
address comment

mridulm commented:
Merged to master, thanks or fixing this @JiexingLi !
Thanks for looking into this @HyukjinKwon :-)
